mantidproject/mantid
# Run information for co-added runs in MuonAnalysis

Fixes the run information displayed in the MuonAnalysis interface when several runs are co-added.

Corrects each point raised in the issue:
- "Runs" deals with non-consecutive ranges and gaps
- "Sample temperature" and "sample magnetic field" are shown as a range if they are different for each run
- "Average temperature" is derived from all the runs
- "Start" and "End" are the earliest start and latest end

Previously these fields showed data which applied only to the first run.

The new functionality is in `MuonAnalysisHelper` rather than the main MuonAnalysis class, so it was possible to add unit tests. Each new piece of functionality should have a test.

**To test:**
_Code review:_ 
Ensure the tests cover the new functionality

_Functional tests:_
Files `MUSR15189.nxs` to `MUSR15193.nxs` can be found in the DocTest folder.
- Start MuonAnalysis interface
- Load a selection of these files by typing a range of run numbers into the box
- Verify "runs" is shown correctly
- Check the start and end times, and range of magnetic fields
- Try a sequence of files with a gap e.g. 15189-91, 15193
- To test the sample temperature/average temperature, different files are needed. Try MUSR24580-1 in the training data (can be found at `\\olympic\babylon5\Scratch\TomPerkins\Data`) to see a range of sample temperatures, and check the average temperature is derived from both files.
- Test the above when you have the default instrument in Mantid (Help/First time setup) set to a different instrument than the one selected in the interface

Fixes #13738 

The [release notes](https://github.com/mantidproject/mantid/blob/13738_Run_info_MuonAnalysis/docs/source/release/v3.7.0/muon.rst) have been updated.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
